src/doughnut/signingMethods/0/index.js

```diff
@@ -2,12 +2,19 @@
  * V0 Signing Method - Schnorrkel *
  **********************************/
 const {
+  cryptoWaitReady,
   schnorrkelSign,
   schnorrkelVerify,
 } = require("@polkadot/util-crypto");
 
+// Wrap schnorrkel signing to await wasm crypto readiness
+async function sign(payload, keypair) {
+    await cryptoWaitReady();
+    return schnorrkelSign(payload, keypair);
+}
+
 module.exports = {
-  sign: schnorrkelSign,
+  sign: sign,
   verify: schnorrkelVerify,
```

Comment:
Does schnorrkel verify not require the same treatment? (same question for signing method 1)

Reply:
Yes, very good point! updated.